Deployment/Jenkins/master/parent-instance.groovy

```diff
@@ -0,0 +1,303 @@
+                        "ParameterKey": "ElbTemplate",
+                        "ParameterValue": "${env.ElbTemplate}"
+                      },
+                      {
+                        "ParameterKey": "IamRoleTemplate",
+                        "ParameterValue": "${env.IamRoleTemplate}"
+                      },
+                      {
+                        "ParameterKey": "SecurityGroupTemplate",
+                        "ParameterValue": "${env.SecurityGroupTemplate}"
+                      },
+                      {
+                        "ParameterKey": "TargetVPC",
+                        "ParameterValue": "${env.TargetVPC}"
+                      },
+                      {
+                      "ParameterKey": "BackupCronURL",
+                      "ParameterValue": "${env.BackupCronURL}"
+                      },
+                      {
+                      "ParameterKey": "SubnetId",
+                      "ParameterValue": "${env.SubnetId}"
+                      }
+                    ]
+                   /
+                }
+            }
+        stage ('Prepare AWS Environment') {
+            options {
+                timeout(time: 1, unit: 'HOURS')
+            }
+            steps {
+                withCredentials(
+                    [
+                        [$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: "${AwsCred}", secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'],
+                        sshUserPrivateKey(credentialsId: "${GitCred}", keyFileVariable: 'SSH_KEY_FILE', passphraseVariable: 'SSH_KEY_PASS', usernameVariable: 'SSH_KEY_USER')
+                    ]
+                ) {
+                    sh '''#!/bin/bash
+                        echo "Attempting to delete any active ${CfnStackRoot} stacks... "
+                        aws --region "${AwsRegion}" cloudformation delete-stack --stack-name "${CfnStackRoot}"
+
+                        aws cloudformation wait stack-delete-complete --stack-name ${CfnStackRoot} --region ${AwsRegion}
+                    '''
+                }
+            }
+        }
+        stage ('Launch Jenkins Master Stack') {
+            options {
+                timeout(time: 1, unit: 'HOURS')
+            }
+            steps {
+                withCredentials(
+                    [
+                        [$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: "${AwsCred}", secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'],
+                        sshUserPrivateKey(credentialsId: "${GitCred}", keyFileVariable: 'SSH_KEY_FILE', passphraseVariable: 'SSH_KEY_PASS', usernameVariable: 'SSH_KEY_USER')
+                    ]
+                ) {
+                    sh '''#!/bin/bash
+                        echo "Attempting to create stack ${CfnStackRoot}..."
+                        aws --region "${AwsRegion}" cloudformation create-stack --stack-name "${CfnStackRoot}" \
```

Comment:
See note in agent-pipeline about avoiding (potentially disastrous) collisions.

If these are to be used for prouction stacks, might be worth adding `${BUILD_NUMBER}`  (see _[Jenkins Standard Envs](https://wiki.jenkins-ci.org/display/JENKINS/Building+a+software+project#Buildingasoftwareproject-JenkinsSetEnvironmentVariables)_). This should make every run's resultant stacks unique and avoid disasters like nuking production if someone accidentally hits rebuild/replay on the most recent run.

Reply:
It appears `${CfnStackRoot}-${BUILD_NUMBER}` will provide the following in the job console output but doesn't actually create the `jdmtest00-39` job but `jdmtest00` instead. 
```
[jenkinsfile] Running shell script
Attempting to create stack jdmtest00-39...
{
    "StackId": "arn:aws:cloudformation:us-east-1:420895463031:stack/jdmtest00/aba4c1d0-b21c-11e8-9d7e-500c288f18d1"
}
```

Reply:
Ok, that would make sense if you had:

~~~
                        echo "Attempting to delete any active ${CfnStackRoot}-${BUILD_NUMBER} stacks... "
                        aws --region "${AwsRegion}" cloudformation delete-stack --stack-name "${CfnStackRoot}"
                         aws cloudformation wait stack-delete-complete --stack-name ${CfnStackRoot} --region ${AwsRegion}
~~~

but not:

~~~
                        echo "Attempting to delete any active ${CfnStackRoot}-${BUILD_NUMBER} stacks... "
                        aws --region "${AwsRegion}" cloudformation delete-stack --stack-name "${CfnStackRoot}-${BUILD_NUMBER}"
                         aws cloudformation wait stack-delete-complete --stack-name "${CfnStackRoot}-${BUILD_NUMBER}" --region ${AwsRegion}
~~~

With what's currently in the reviewable part of the PR, I can't see what you have.

Reply:
See comment on your [_branch-commit_](https://github.com/murphyj21/cfn-jenkins/commit/cfcba8736bf819a8dd04a4963a041cc76afe3561#diff-f8c0569e877046b766dfbda11b8901ebR292).

Reply:
I'm a flipping idiot...stand by

Reply:
Ok, done and working properly.